ivy/functional/frontends/tensorflow/general_functions.py

```diff
@@ -80,6 +81,23 @@ def fill(dims, value, name=None):
     return ivy.full(dims, value)
 
 
+@to_ivy_arrays_and_back
+def foldl(fn, elems, initializer=None):
+    try:
+        iter(elems)
+    except TypeError:
+        raise TypeError("elems must be an iterable object")
```

Comment:
Nit: maybe use `ivy.utils.exceptions.IvyException` :thinking:?

Reply:
That's what I went with originally but saw that no where else in the file was an ivy.exception used or thrown, will change it as requested.